agent/agent.go

```diff
@@ -207,7 +207,7 @@ func (a *Agent) run(ctx context.Context) {
 
 			// select a session registration delay from backoff range.
 			delay := time.Duration(rand.Int63n(int64(backoff)))
-			session = newSession(ctx, a, delay)
+			session = newSession(ctx, a, delay, session.sessionID)
```

Comment:
PTAL at this line, because I'm not sure if this is the behavior we want. this sends the old session ID to skip rate limiting EVERY TIME the session dies.